launch/launch/actions/execute_process.py

```diff
@@ -183,13 +188,15 @@ def __init__(
         self.__cmd = [normalize_to_list_of_substitutions(x) for x in cmd]
         self.__name = name if name is None else normalize_to_list_of_substitutions(name)
         self.__cwd = cwd if cwd is None else normalize_to_list_of_substitutions(cwd)
-        self.__env = None  # type: Optional[List[Tuple[List[Substitution], List[Substitution]]]]
-        if env is not None:
-            self.__env = []
-            for key, value in env.items():
-                self.__env.append((
-                    normalize_to_list_of_substitutions(key),
-                    normalize_to_list_of_substitutions(value)))
+        if env is None:
+            env = dict(os.environ)
```

Comment:
@ivanpauno this patch is effectively changing behavior: when no `env` is provided, it is no longer inherited but forced to be the one it was at construction time.

Reply:
Good catch, I will change it again.